modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySpiLogTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.discovery.tcp;
+
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.LinkedHashSet;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.spi.discovery.tcp.internal.TcpDiscoveryNode;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.testframework.GridTestUtils.waitForCondition;
+
+/**
+ * Class for {@link TcpDiscoverySpi} logging tests.
+ */
+public class TcpDiscoverySpiLogTest extends GridCommonAbstractTest {
+    /**
+     * Listener log messages
+     */
+    private static ListeningTestLogger testLog;
+
+    /** */
+    private static final ThreadLocal<TcpDiscoverySpi> nodeSpi = new ThreadLocal<>();
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        TcpDiscoverySpi spi = nodeSpi.get();
+
+        if (spi == null)
+            spi = new TcpDiscoverySpi();
+        else
+            nodeSpi.set(null);
+
+        cfg.setDiscoverySpi(spi);
+
+        cfg.setGridLogger(testLog);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        testLog = new ListeningTestLogger(log);
```

Comment:
If we assing the log listeners only for the checking node, might be moved to `getConfiguration()` or to certain tests. Would make fewer functions.